### PR TITLE
Add Array.prototype.reduce

### PR DESCRIPTION
**Pull request by cpq**
Added Date test 2068 to ignore list.

### CODE REVIEW

**Comment by cpq** [comment]
Rebased against master. PTAL

File: src/stdlib.c [hunk 1]
```diff
@@ -251,6 +251,23 @@ Array.prototype.lastIndexOf = function(a, b) {
   return -1;
 };
 ));
+
+  v7_exec(v7, &res, STRINGIFY(
+    Array.prototype.reduce = function(a, b) {
+      var f = 0; var r = b; var t = this;
+      if (typeof(a) != 'function') {
+        throw new TypeError('oops');
```
Comment (mmikulicic) [inline src/stdlib.c]: `a + 'is not a function'`

Comment (cpq) [inline src/stdlib.c]: Done.
Also, increased a limit in `i_populate_local_vars()` to make jquery compile

File: src/stdlib.c [hunk 2]
```diff
@@ -251,6 +251,23 @@ Array.prototype.lastIndexOf = function(a, b) {
   return -1;
 };
 ));
+
+  v7_exec(v7, &res, STRINGIFY(
+    Array.prototype.reduce = function(a, b) {
+      var f = 0; var r = b; var t = this;
```
Comment (mmikulicic) [inline src/stdlib.c]: let's just mutate the argument `b`.
Let's also use `this`. Minification doesn't pay off here. 13 bytes for decl to save 9 bytes. `this` is already passed, `t` takes extra space on the environment and AST-code is longer (`AST_THIS` is a special opcode)

**Comment by mmikulicic** [comment]
LGTM, modulo minor inline comment

File: src/stdlib.c (hunk 2 above)
Comment (cpq) [inline src/stdlib.c]: Done

**Pull request merged**